tests/e2e/faultinjection/duplicate_node_id.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package faultinjection
+
+import (
+	"context"
+
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/api/info"
+	cfg "github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/tests/e2e"
+	"github.com/ava-labs/avalanchego/tests/fixture/testnet"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var _ = ginkgo.Describe("Duplicate node handling", func() {
+	require := require.New(ginkgo.GinkgoT())
+
+	ginkgo.It("should ensure that a given Node ID (i.e. staking keypair) can be used at most once on a network", func() {
+		network := e2e.Env.GetNetwork()
+		nodes := network.GetNodes()
+
+		ginkgo.By("creating new node")
+		node1 := e2e.AddTemporaryNode(network, testnet.FlagsMap{})
+		e2e.WaitForHealthy(node1)
```

Comment:
Just so I understand better why here is `WaitForHealthy` takes a node but below it is a method on node. Just generally do we need both ways to do this? Regardless they should be the same through the test. I could be missing something though.

Reply:
OK I think it is strange that we have two methods with the same name and different functionality and signatures. IMO `WaitForHealthy` should always take a ctx and then allow the test writer the choice of timeout. If a method infers a set timeout that should be something else?